### ISSUE TITLE
ISO/TS 32001 and ISO 32000-2:2020 Amendment 1 are very confusing

### ISSUE THREAD

**Issue by petervwyatt** [pdf-association/pdf-issues#236]
* Adobe PDF 1.5 Table 8.94 says **DigestMethod** was Optional with a default of _MD5_.
* Adobe PDF 1.7 Table 8.103 says **DigestMethod** was Optional with a default of _MD5_, but Implementation Note 144 says it was really required by Adobe’s implementation.
* ISO 32000-1:2008 Table 253 says **DigestMethod** was "(Optional; PDF 1.5 required)" but with no default value defined. Only _MD5_ and _SHA1_ values are defined.
* ISO 32000-2:2017 Table 256 says **DigestMethod** was Optional but with a default value of MD5 for PDF 1.5-1.7 (so no default stated for PDF 2.0). It also introduced the extra values _SHA256_, _SHA384_, _SHA512_ and _RIPEMD160_.
* ISO 32000-2:2020 Table 256 changed **DigestMethod** to Required and that PDF 1.5-1.7 had a default value of _MD5_ (so no default stated for PDF 2.0)
* The resolution of [Errata #117](https://github.com/pdf-association/pdf-issues/issues/117) was to change **DigestMethod** to optional and indicate as deprecated in PDF 2.0 (as shown in a modified [Table 256](https://pdf-issues.pdfa.org/32000-2-2020/clause12.html#Table256)).
* [Errata #117](https://github.com/pdf-association/pdf-issues/issues/117) was approved by WG8 in April 2022 and is included in ISO 32000-2:2020 Amendment 1 (XFDF).
* ISO/DTS 32001 clause 5.1.3 added new values to **DigestMethod** and defined a new default value (_SHA256_) for PDF 2.x
* In WG8 document N0953 against ISO/DTS 32001, comment BE-006 highlighted this fact and proposed removal, but the PL at the time resolved with “Not accepted. As this TS explicitly targets ISO 32000-2:2020 & the signature reference dictionary is not deprecated for ISO 32000-2:2020, PL believes these update hash algorithms should be allowed for use in a signature reference dictionary.”. There were no further reviews of ISO/DTS 32001 prior to submission to ISO CS.
* IS/TS 32001 is now [published by ISO](https://www.iso.org/standard/45874.html) with the additional new values for **DigestMethod** and defines a default value for PDF 2.0 for the deprecated **DigestMethod** key:
    > (PDF 2.x) In addition to those values previously defined, this value may be one of _SHA3-256_, _SHA3-384_, _SHA3-512_ or _SHAKE256_. Default value for PDF 2.0: _SHA256_ (PDF 2.0).

It appears that **DigestMethod** has no function in ISO 32000-x, so the change in ISO/TS 32001 is effectively a “no op”, but is nonetheless extremely confusing. This was originally noticed by N0954 BE-006 and then re-discovered because of trying to model ISO/TS 32001 logically into the Arlington PDF Model.

It would be good to clarify the situation.

**Comment by MatthiasValvekens** [comment]
Since I submitted the comment (BE-006) that Peter refers to, I can perhaps shed some light on why it was disposed the way it is:

> In WG8 document N0953 against ISO/DTS 32001, comment BE-006 highlighted this fact and proposed removal, but the PL at the time resolved with “Not accepted. As this TS explicitly targets ISO 32000-2:2020 & the signature reference dictionary is not deprecated for ISO 32000-2:2020, PL believes these update hash algorithms should be allowed for use in a signature reference dictionary.”. There were no further reviews of ISO/DTS 32001 prior to submission to ISO CS.

The whole conversation around #117 unfortunately happened when the ISO 32001 project was already running up against the ISO clock, and we couldn't easily strike the **DigestMethod** clause at that point in the process without putting the schedule in jeopardy.

Around the time the DTS ballot for 32001 went out, this was discussed over email in the Encryption & Digital Signatures DG, and the conclusion of that conversation was more or less the following: given that the extension of **DigestMethod** was functionally a no-op, the least bad option was to leave the text as-is and remove the vestigial clause from 32001 at some later point in time.

The PL and I agreed that I would submit a comment on the ballot to request the change anyway, just to have it on record (and we'd have something to point to when 32001 is next revised).


TL;DR: Clause 5.1.3 should be stricken from 32001 at the next opportunity. The only reason why that didn't happen earlier is because the deprecation of **DigestMethod** was decided at a time when it was no longer practical to make technical changes to the 32001 draft.

**Comment by petervwyatt** [comment]
@MatthiasValvekens - as Chair, could you please have the DigSig TWG review this errata? 
If there is agreement in the DigSig TWG, I am then happy to publish their resolution as an errata against ISO/TS 32001

**Comment by faceless2** [comment]
Agreed - `DigestMethod` is unused and should go.

Back when 1.7 first came out the `DocMDP` validation process was supposed involve storing the digest in the file, then recalculating it on load (according to the _Computation of Object Digests_ algorithm from the 1.7 appendix). I can tell you we tried long and hard to do that, on the absolute simplest documents, and at no point could we make it work - I presume an omission from the algorithm. So the process has always been calculate on the original version, calculate on the new version and compare for equality. `DigestMethod` never came into it.

**Comment by mkl-public** [comment]
>Back when 1.7 first came out the `DocMDP` validation process was supposed involve storing the digest in the file, then recalculating it on load

Strictly speaking using that object digest for MDP was the state of the PDF 1.5 reference, already in PDF 1.6 that `DigestValue` became optional and applications were expected to _compare the signed and current versions of the document to see whether there have been modifications to any objects that are not permitted by the transform parameters_.

**Comment by petervwyatt** [comment]
Approved by WG8 in Prague.
